okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/metrics/HostMetricsRegistry.java

```diff
@@ -20,34 +20,48 @@
 import com.google.common.cache.CacheLoader;
 import com.google.common.cache.LoadingCache;
 import com.palantir.logsafe.UnsafeArg;
+import com.palantir.remoting3.clients.ImmutablesStyle;
 import com.palantir.tritium.metrics.registry.TaggedMetricRegistry;
+import java.util.Collection;
 import java.util.concurrent.TimeUnit;
+import org.immutables.value.Value;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public final class HostMetricsRegistry {
 
     private static final Logger log = LoggerFactory.getLogger(HostMetricsRegistry.class);
 
-    private final LoadingCache<String, HostMetrics> hostMetrics;
+    private final LoadingCache<ServiceAndHost, HostMetrics> hostMetrics;
 
-    public HostMetricsRegistry(TaggedMetricRegistry registry, String serviceName) {
+    public HostMetricsRegistry(TaggedMetricRegistry registry) {
         this.hostMetrics = CacheBuilder.newBuilder()
                 .maximumSize(1_000)
                 .expireAfterAccess(1, TimeUnit.DAYS)
-                .build(new CacheLoader<String, HostMetrics>() {
+                .build(new CacheLoader<ServiceAndHost, HostMetrics>() {
                     @Override
-                    public HostMetrics load(String hostname) throws Exception {
-                        return new HostMetrics(registry, serviceName, hostname);
+                    public HostMetrics load(ServiceAndHost serviceAndHost) throws Exception {
+                        return new HostMetrics(registry, serviceAndHost.serviceName(), serviceAndHost.hostname());
                     }
                 });
     }
 
-    public void record(String hostname, int statusCode) {
+    public void record(String serviceName, String hostname, int statusCode) {
         try {
-            hostMetrics.getUnchecked(hostname).record(statusCode);
+            hostMetrics.getUnchecked(ImmutableServiceAndHost.of(serviceName, hostname)).record(statusCode);
         } catch (Exception e) {
             log.warn("Unable to record metrics for host", UnsafeArg.of("hostname", hostname));
         }
     }
+
+    public Collection<HostMetrics> getMetrics() {
```

Comment:
possibly worth exposing the map here. means we can't record things finer grained that the service/host level.